builder.go

```diff
@@ -215,6 +215,17 @@ type Stage struct {
 
 func NewStages(node *parser.Node, b *Builder) (Stages, error) {
 	var stages Stages
+	var allDeclaredArgs []string
+	for _, root := range SplitBy(node, command.Arg) {
+		argNode := root.Children[0]
+		if argNode.Value == command.Arg {
+			// extract declared variable
+			s := strings.SplitN(argNode.Original, " ", 2)
+			if len(s) == 2 && (strings.ToLower(s[0]) == strings.ToLower("ARG")) {
```

Comment:
strings.ToLower("ARG") versus "arg"?

Reply:
Yes, switched this to an already defined constant here https://github.com/openshift/imagebuilder/blob/master/dockerfile/command/command.go#L7